manifests/globals.pp

```diff
@@ -38,17 +38,30 @@
   }
 
   # Setup of the repo only makes sense globally, so we are doing it here.
-  if $manage_package_repo {
-    if $use_enterprise_repo == true and $version == undef {
-      fail('You must set mongodb::globals::version when mongodb::globals::use_enterprise_repo is true')
+  case $facts['os']['family'] {
+    'RedHat', 'Linux', 'Suse': {
+      class { 'mongodb::repo':
+        ensure              => present,
+        version             => pick($version, '3.6'),
+        use_enterprise_repo => $use_enterprise_repo,
+        repo_location       => $repo_location,
+        proxy               => $repo_proxy,
+      }
```

Comment:
Shouldn't that be wrapped into `manage_package_repo` as well?

Reply:
I think you are right, I did this first because without this repo, mongodb won't get installed. Before the module would have use the package from epel.

Reply:
I think this will break test units.

Reply:
I just push the change.

Reply:
Can you extract the condition outside the case statement as well?

Reply:
I don't understand I want this only for redhat.

Reply:
You are checking `if $manage_package_repo` twice, once for RedHat and once for default. You can wrap both inside `if $manage_package_repo` instead of checking it twice.

Reply:
Hmm yea you can revert that :(